### ISSUE TITLE
[Regression] supply crashes at non-aab project

### ISSUE THREAD

**Issue by Nkzn** [fastlane/fastlane#13005]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information

- Breaking version: `2.101.0`
- Last working version: `2.95.0` (probably `2.100.1`)

### Regression Description

Before `2.101.0`, `supply/uploader.rb` had below lines.

https://github.com/fastlane/fastlane/blob/56e8020a942b86999df82e0d681502f13dec70d3/supply/lib/supply/uploader.rb#L27

https://github.com/fastlane/fastlane/blob/56e8020a942b86999df82e0d681502f13dec70d3/supply/lib/supply/uploader.rb#L148-L158

At the above environment, we could write below `supply` call.

```ruby
supply(
  track: "beta",
  apk: APK_PATH,
  package_name: "jp.agri_note.android",
  skip_upload_metadata: true
)
```

This worked with no `skip_upload_aab` setting, because `upload_bundles` returned only `nil`, and uploader.rb#L27 did nothing.

However, `2.101.0` has below lines.

https://github.com/fastlane/fastlane/blob/3bc37e5dbd4fc319ce0688e582fe0e9aff31a4b2/supply/lib/supply/uploader.rb#L28

https://github.com/fastlane/fastlane/blob/3bc37e5dbd4fc319ce0688e582fe0e9aff31a4b2/supply/lib/supply/uploader.rb#L154-L160

**At the above environment, `concat` at uploader.rb#L28 produces error if we don't write both `skip_upload_aab` and `aab` in `supply`, because `upload_bundles` returns `nil`.**

Probably, non aab users can reproduce this error.

#### workaround

Add `skip_upload_aab: true` in `supply` call.

```diff
 supply(
   track: "beta",
   apk: APK_PATH,
   package_name: "jp.agri_note.android",
   skip_upload_metadata: true,
+  skip_upload_aab: true
 )
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>bundler: failed to load command: fastlane (/Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/bin/fastlane)
TypeError: [!] no implicit conversion of nil into Array
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/supply/lib/supply/uploader.rb:28:in `concat'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/supply/lib/supply/uploader.rb:28:in `perform_upload'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/actions/upload_to_play_store.rb:21:in `run'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:33:in `block (2 levels) in parsing_binding'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/gems/fastlane-2.101.0/bin/fastlane:20:in `<top (required)>'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `load'
  /Users/jenkins/builds/0e6a6a18/0/agrinote/agri-note-android-v2/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `<top (required)>'</pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                             |
| --------------------------- | ------------------------------------------------- |
| OS                          | 10.13.6                                           |
| Ruby                        | 2.4.1                                             |
| Bundler?                    | true                                              |
| Git                         | git version 2.17.1                                |
| Installation Source         | ~/dev/anapp/vendor/bundle/ruby/2.4.0/bin/fastlane |
| Host                        | Mac OS X 10.13.6 (17G65)                          |
| Ruby Lib Dir                | ~/.anyenv/envs/rbenv/versions/2.4.1/lib           |
| OpenSSL Version             | OpenSSL 1.1.0f  25 May 2017                       |
| Is contained                | false                                             |
| Is homebrew                 | false                                             |
| Is installed via Fabric.app | false                                             |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/       |
| Xcode Version               | 9.4.1                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | ja_JP.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.27.0"

default_platform :android

platform :android do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  end

  desc "Deploy a new version to the Google Play"
  lane :deploy_beta_an do
    APK_PATH = "AgriNote/build/outputs/apk/an/release/AgriNote-an-release.apk"

    gradle(task: "assembleAnRelease")
    supply(
      track: "beta",
      apk: APK_PATH,
      package_name: "jp.agri_note.android",
      skip_upload_metadata: true
    )
  end

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    chatwork(
      message: "error occured!",
      roomid: "XXXXXXXX",
      success: false
    )

    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is sent or shared. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file "" # Path to the json secret file - Follow https://github.com/fastlane/supply#setup to get one
package_name "" # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.101.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                              | Version | Update-Status |
| ----------------------------------- | ------- | ------------- |
| fastlane-plugin-get_android_version | 0.3.5   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                 | Version      |
| ----------------------------------- | ------------ |
| did_you_mean                        | 1.1.0        |
| bundler                             | 1.16.3       |
| io-console                          | 0.4.6        |
| CFPropertyList                      | 3.0.0        |
| public_suffix                       | 2.0.5        |
| addressable                         | 2.5.2        |
| rubyzip                             | 1.2.1        |
| apktools                            | 0.7.2        |
| atomos                              | 0.1.2        |
| babosa                              | 1.0.2        |
| claide                              | 1.0.2        |
| colored                             | 1.2          |
| colored2                            | 3.1.2        |
| highline                            | 1.7.10       |
| commander-fastlane                  | 4.4.6        |
| declarative                         | 0.0.10       |
| declarative-option                  | 0.1.0        |
| unf_ext                             | 0.0.7.5      |
| unf                                 | 0.1.4        |
| domain_name                         | 0.5.20180417 |
| dotenv                              | 2.5.0        |
| emoji_regex                         | 0.1.1        |
| excon                               | 0.62.0       |
| multipart-post                      | 2.0.0        |
| faraday                             | 0.15.2       |
| http-cookie                         | 1.0.3        |
| faraday-cookie_jar                  | 0.0.6        |
| faraday_middleware                  | 0.12.2       |
| fastimage                           | 2.1.3        |
| gh_inspector                        | 1.1.3        |
| jwt                                 | 2.1.0        |
| little-plugger                      | 1.1.4        |
| multi_json                          | 1.13.1       |
| logging                             | 2.2.2        |
| memoist                             | 0.16.0       |
| os                                  | 0.9.6        |
| signet                              | 0.8.1        |
| googleauth                          | 0.6.2        |
| httpclient                          | 2.8.3        |
| mime-types-data                     | 3.2016.0521  |
| mime-types                          | 3.1          |
| uber                                | 0.1.0        |
| representable                       | 3.0.4        |
| retriable                           | 3.1.2        |
| google-api-client                   | 0.23.4       |
| json                                | 2.1.0        |
| mini_magick                         | 4.5.1        |
| multi_xml                           | 0.6.0        |
| plist                               | 3.4.0        |
| security                            | 0.1.3        |
| naturally                           | 2.2.0        |
| simctl                              | 1.6.5        |
| slack-notifier                      | 2.3.2        |
| terminal-notifier                   | 1.8.0        |
| unicode-display_width               | 1.4.0        |
| terminal-table                      | 1.8.0        |
| tty-screen                          | 0.6.5        |
| tty-cursor                          | 0.6.0        |
| tty-spinner                         | 0.8.0        |
| word_wrap                           | 1.0.0        |
| nanaimo                             | 0.2.6        |
| xcodeproj                           | 1.5.9        |
| rouge                               | 2.0.7        |
| xcpretty                            | 0.2.8        |
| xcpretty-travis-formatter           | 1.0.0        |
| fastlane-plugin-get_android_version | 0.3.5        |
</details>

*generated on:* **2018-08-01**
</details>

**Comment by Nkzn** [comment]
#12998 fixes this issue.

**Comment by joshdholtz** [comment]
Fixed in #12998 and will be released in `2.101.1` shortly
